packages/mui-system/src/Stack/createStack.tsx

```diff
@@ -135,7 +135,7 @@ export const style = ({ ownerState, theme }: StyleFunctionProps) => {
         return { gap: getValue(transformer, propValue) };
       }
       return {
-        '& > :not(style) + :not(style)': {
+        '& > :not(style):not(style)': {
```

Comment:
There is a subtle difference b/w `:not(style) + :not(style)` and `:not(style):not(style)`.

`:not(style):not(style)`  targets any element that is a direct child of an parent and the element is not a <style> element. The :not(style):not(style) part ensures that the selected element is not a <style> element.

`:not(style) + :not(style)` targets any element that is a direct child of an parent and the element is immediately followed by another element that is not a <style> element.

If you look at this below codepen where `:not(style) + :not(style)` is applied to parent class, you can see styles are applied to all elements except 1st child.

https://codepen.io/sai6855/pen/abQbmWp

Now if you look at below codepen where `:not(style):not(style)` is applied to parent class, you can see styles are applied to all children including 1st. 

https://codepen.io/sai6855/pen/WNYNGMV

Reply:
Hey @sai6855! A couple questions 😊:

- I think `:not(style):not(style)` is equivalent to `:not(style)`, right? If that's the case, I think we should use `:not(style)`
- Do you know why it was originally `:not(style) + :not(style)`? doesn't seem like a mistake or typo 🤔 I'm wondering if there was a reason to use that selector, and if so, why. I ask this to avoid potential regressions.

Reply:
> I think :not(style):not(style) is equivalent to :not(style), right? If that's the case, I think we should use :not(style)

Yup you are right, removed one redundant one `not(style)` in this [commit](https://github.com/mui/material-ui/pull/37525/commits/ddae1dbce30d1bad4dbb5641e38575cb95a61a85) .

>Do you know why it was originally :not(style) + :not(style)? doesn't seem like a mistake or typo 🤔 I'm wondering if there was a reason to use that selector, and if so, why. I ask this to avoid potential regressions.

TBH i'm not really sure why `:not(style) + :not(style)` is used previously. based on commit history maybe @mnajdova might know

Reply:
> TBH i'm not really sure why :not(style) + :not(style) is used previously

This code adds top margins to all but the first stack items. With your change, all child elements will have a margin applied, which is not desired (compare https://deploy-preview-37525--material-ui.netlify.app/material-ui/react-stack/#basics to https://mui.com/material-ui/react-stack/#basics).

What I don't know is why the other margin values (sides and bottom) are also affected. Perhaps @mnajdova (or @siriwatknp) would remember.

Reply:
> I think :not(style):not(style) is equivalent to :not(style), right? If that's the case, I think we should use :not(style)

@DiegoAndai The specificity is different https://specificity.keegan.st/

<img width="806" alt="Screenshot 2023-07-01 at 14 19 35" src="https://github.com/mui/material-ui/assets/3165635/6e621012-c381-4ab3-9c85-cf26bb32e3ce">

Which is necessary because of

<img width="763" alt="Screenshot 2023-07-01 at 14 20 15" src="https://github.com/mui/material-ui/assets/3165635/4f979517-cc4c-4be0-8d48-b66d028c43d7">